hal/src/nRF52840/platform_radio_stack.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform_radio_stack.h"
+#include "ota_flash_hal_impl.h"
+#include "flash_mal.h"
+#include "check.h"
+#include "nrf_sdm.h"
+#include "nrf_mbr.h"
+
+int platform_radio_stack_fetch_module_info(hal_system_info_t* sys_info, bool create) {
+    for (int i = 0; i < sys_info->module_count; i++) {
+        hal_module_t* module = sys_info->modules + i;
+        if (memcmp(&module->bounds, &module_radio_stack, sizeof(module_radio_stack))) {
```

Comment:
Doesn't this need to check all the module bounds before deciding to create? At it is it checks just the first.

Reply:
I'm not sure I get this. Will we get multiple modules in `sys_info` with exactly the same module bounds that match `module_radio_stack`?